web/containers/admin/AssessmentPage/index.js

```diff
@@ -67,30 +69,88 @@ class AssessmentPageContainer extends React.Component {
     });
   }
 
+  switchSortType = (newSortType) => {
+    this.setState({
+      sortType: newSortType,
+      selectedApplicantId: null,
+    });
+  };
+
+  sortApplicants = (sortType, applicants) => {
+    switch (sortType) {
+      case 'name':
+        return applicants.sort((a, b) => {
+          if (!a.lastName) return -1;
+          if (!b.lastName) return 1;
+          if (a.lastName.toUpperCase() > b.lastName.toUpperCase()) return 1;
+          if (a.lastName.toUpperCase() < b.lastName.toUpperCase()) return -1;
+          return 0;
+        });
+      case 'email':
+        return applicants.sort((a, b) => {
+          if (!a.email) return -1;
+          if (!b.email) return 1;
+          if (a.email.toUpperCase() > b.email.toUpperCase()) return 1;
+          if (a.email.toUpperCase() < b.email.toUpperCase()) return -1;
+          return 0;
+        });
+      case 'score':
+        return applicants.sort((a, b) => {
+          if (!a.score || !a.score.finalScore) return -1;
+          if (!b.score || !b.score.finalScore) return 1;
+          if (a.score.finalScore > b.score.finalScore) return 1;
+          if (a.score.finalScore < b.score.finalScore) return -1;
+          return 0;
+        });
+      case 'timestamp':
+        return applicants.sort((a, b) => {
+          if (!a.timestamp) return -1;
+          if (!b.timestamp) return 1;
+          if (a.timestamp > b.timestamp) return 1;
+          if (a.timestamp < b.timestamp) return -1;
+          return 0;
+        });
+      default:
+        return applicants;
+    }
+  };
+
+  reverseSort = () => {
+    const { reverse } = this.state;
+    const newReverse = !reverse;
+    this.setState({ reverse: newReverse });
+  };
+
   render() {
     const {
       applicantType,
       selectedApplicantId,
+      sortType,
+      reverse,
     } = this.state;
 
     const {
       firestore,
     } = this.props;
-
     const shortInfoCollectionName = applicantCollections[applicantType].shortInfo;
     const applicants = firestore.ordered[shortInfoCollectionName];
-
+    let sortedApplicants = this.sortApplicants(sortType, applicants);
+    if (reverse) sortedApplicants = applicants.reverse();
     if (!isLoaded(applicants)) {
```

Comment:
I didn't catch this in my first review but it looks like switching to mentor or volunteer breaks the page because L137 & L138 should not come before this if statement on L139.